ocaml/lang/security/filenameconcat.yaml

```diff
@@ -0,0 +1,16 @@
+rules:
+  - id: ocamllint-filenameconcat
+    pattern: Filename.concat
+    message: When attacker supplied data is passed to Filename.concat directory traversal attacks might be possible.
```

Comment:
I would at least add a `pattern-not` to exclude hardcoded strings.

Reply:
```suggestion
    patterns:
      - pattern-either:
        - pattern: Filename.concat $STR $STR
        - pattern: Filename.concat $STR "..."
        - pattern: Filename.concat "..." $STR
      - pattern-not: Filename.concat "..." "..."
```

Reply:
```suggestion
    patterns:
      - pattern-either:
        - pattern: Filename.concat $STR $STR
        - pattern: Filename.concat $STR "..."
        - pattern: Filename.concat "..." $STR
      - pattern-not: Filename.concat "..." "..."
    message: When attacker supplied data is passed to Filename.concat directory traversal attacks might be possible.
```

Reply:
```
    patterns:
      - pattern: Filename.concat $STR $STR
      - pattern-not: Filename.concat "..." "..."
```
I think this should be sufficient